LunaDll/Misc/RuntimeHookComponents/RuntimeHookHooks.cpp

```diff
@@ -3831,6 +3849,173 @@ __declspec(naked) void __stdcall runtimeHookBlockPlayerFilter(void)
     }
 }
 
+static unsigned int __stdcall runtimeHookPlayerNPCInteractionCheckInternal(short playerIdx, short npcIdx)
+{
+    // Player's nonpcinteraction flag
+    ExtendedPlayerFields* playerExt = Player::GetExtended(playerIdx);
+
+    if (playerExt->nonpcinteraction)
+    {
+        return 0;
+    }
+
+    // Collision groups
+    ExtendedNPCFields* npcExt = NPC::GetRawExtended(npcIdx);
+
+    if (!gCollisionMatrix.getIndicesCollide(playerExt->collisionGroup,npcExt->collisionGroup)) // Check collision matrix
+    {
+        return 0;
+    }
+
+    return -1;
+}
+
+__declspec(naked) void __stdcall runtimeHookPlayerNPCInteractionCheck(void)
+{
+    __asm {
+        push eax
+
+        push eax                // NPC index
+        movsx eax, word ptr ss:[ebp-0x114]
+        push eax                // Player index
+
+        call runtimeHookPlayerNPCInteractionCheckInternal
+
+        cmp eax, 0 // return value
+        jne continue_collision
+        jmp cancel_collision
+    continue_collision:
+        pop eax
+
+        movsx edi,ax
+        add edi,0x80
+
+        push 0x9A79F8
+        ret
+    cancel_collision:
+        pop eax
+
+        push 0x9ADCDB
+        ret
+    }
+}
+
+static unsigned int __stdcall runtimeHookPlayerNPCCollisionCheckInternal(short playerIdx)
```

Comment:
Is there a reason why there's no collision matrix check in this function?

Reply:
Yes. The code that handles NPCs being solid for the player is part of the same code that handles the rest of the NPC interaction, and thus collision groups will already have been checked in ``runtimeHookPlayerNPCInteractionCheckInternal``.